src/main/src/test/java/org/geoserver/catalog/ResourcePoolTest.java

```diff
@@ -971,4 +980,290 @@ public void testCoverageReaderInputConverterFilePath() throws IOException {
         // File object
         assertTrue(reader.getSource() instanceof File);
     }
+
+    @Test
+    public void testGetDataStoreConcurrency() throws Exception {
+        ResourcePool pool = null;
+        try {
+            final Catalog catalog = getCatalog();
+            DataStoreInfo ds = storeInfo(catalog, "concurrencyTest1");
+            pool =
+                    new ResourcePool(getCatalog()) {
```

Comment:
This anonymous class seems to be doing nothing new? Why not just use ResourcePool?

Reply:
fixed